### PR TITLE
coverage_journeys table updated to get tad information.

### PR DESCRIPTION
**Pull request by Netmisa**
# Description 

This PR update `coverage_journeys` table to get `nb_with_tad` information per journeys.

SQL request should be:
```
SELECT 
	SUM(cj.nb_with_odt) AS with_odt,
	(SUM(cj.nb) - SUM(cj.nb_with_odt))  AS without_odt
FROM
	stat_compiled.coverage_journeys cj
WHERE
	cj.request_date >= :start_date
	AND cj.request_date < :end_date
	AND cj.region_id = :coverage
	AND cj.is_internal_call = 0
GROUP BY
	cj.region_id
```

### CODE REVIEW

File: migrations/alembic/versions/2dc7abaa8047_add_nb_with_tad_column_from_coverage_.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import sqlalchemy as sa
+import config
+
+schema = config.db['schema']
+
+def upgrade():
+    op.add_column('coverage_journeys', sa.Column('nb_with_odt', sa.BigInteger(), nullable=True), schema=schema)
+
+def downgrade():
+    op.drop_column('coverage_journeys', 'nb_with_odt', schema=schema)
```
Comment (datanel) [inline migrations/alembic/versions/2dc7abaa8047_add_nb_with_tad_column_from_coverage_.py]: have you ever downgraded after inserting in database?

Comment (Netmisa) [inline migrations/alembic/versions/2dc7abaa8047_add_nb_with_tad_column_from_coverage_.py]: Yes, it works

File: migrations/alembic/versions/2dc7abaa8047_add_nb_with_tad_column_from_coverage_.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+down_revision = '51007d933cba'
+
+from alembic import op
+import sqlalchemy as sa
+import config
+
+schema = config.db['schema']
+
+def upgrade():
+    op.add_column('coverage_journeys', sa.Column('nb_with_odt', sa.BigInteger(), nullable=True), schema=schema)
```
Comment (nberard) [inline migrations/alembic/versions/2dc7abaa8047_add_nb_with_tad_column_from_coverage_.py]: u can remove nullable

File: analyzers/coverage_journeys.py [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
     @staticmethod
     def get_tuples_from_stat_dict(stat_dict):
+        nb_journey_with_tad = 0
         journeys = stat_dict.get('journeys', [])
+
+        for journey in journeys:
+            for sections in journey.get('sections', []):
+                if sections['type'] == 'on_demand_transport':
+                    nb_journey_with_tad += 1
+                    continue
```
Comment (nberard) [inline analyzers/coverage_journeys.py]: continue skip to next section not journey ?

Comment (Netmisa) [inline analyzers/coverage_journeys.py]: Exact, I change this to `break`. Thanks

File: migrations/alembic/versions/2dc7abaa8047_add_nb_with_tad_column_from_coverage_.py (hunk 2 above)
Comment (Netmisa) [inline migrations/alembic/versions/2dc7abaa8047_add_nb_with_tad_column_from_coverage_.py]: If we change this we will have problem to set older values.

File: analyzers/coverage_journeys.py [hunk 4]
```diff
@@ -1,11 +1,21 @@
...
+        nb_journey_with_tad = 0
         journeys = stat_dict.get('journeys', [])
+
+        for journey in journeys:
+            for sections in journey.get('sections', []):
+                if sections['type'] == 'on_demand_transport':
+                    nb_journey_with_tad += 1
+                    break
+
+
```
Comment (datanel) [inline analyzers/coverage_journeys.py]:
```python
for journey in journeys:
    if any(sections['type'] == 'on_demand_transport' for section in journey.get('sections', [])):
        nb_journey_with_tad += 1
```

 I think this should work

**Pull request merged**